tests/test_cleanup_kubernetes_jobs.py

```diff
@@ -0,0 +1,184 @@
+    with mock.patch(
+        'paasta_tools.cleanup_kubernetes_jobs.KubeClient',
+        return_value=mock_kube_client,
+        autospec=True,
+    ), mock.patch(
+        'paasta_tools.cleanup_kubernetes_jobs.get_services_for_cluster',
+        return_value={('service', 'instance-1'), ('service', 'instance-2')},
+        autospec=True,
+    ), mock.patch(
+        'paasta_tools.cleanup_kubernetes_jobs.notify',
+        autospec=True,
+    ) as mock_notify:
+        mock_notify.__enter__ = mock.Mock(return_value=(mock.Mock(), None))
+        mock_notify.__exit__ = mock.Mock(return_value=None)
+        cleanup_unused_apps('soa_dir', kill_threshold=1, force=False)
+        assert fake_deployment in mock_kube_client.deployments.list_namespaced_deployment(namespace).items
+        assert fake_stateful_set in mock_kube_client.deployments.list_namespaced_stateful_set(namespace).items
+
+
+def test_cleanup_unused_apps_dont_kill_everything():
+    mock_kube_client = MockKubeClient()
+    namespace = 'paasta'
+    mock_kube_client.deployments.create_namespaced_deployment(namespace, fake_deployment)
+    with mock.patch(
+        'paasta_tools.cleanup_kubernetes_jobs.KubeClient',
+        return_value=mock_kube_client,
+        autospec=True,
+    ), mock.patch(
+        'paasta_tools.cleanup_kubernetes_jobs.get_services_for_cluster',
+        return_value={},
+        autospec=True,
+    ), mock.patch(
+        'paasta_tools.cleanup_kubernetes_jobs.notify',
+        autospec=True,
+    ) as mock_notify:
+        mock_notify.__enter__ = mock.Mock(return_value=(mock.Mock(), None))
+        mock_notify.__exit__ = mock.Mock(return_value=None)
+        with raises(DontKillEverythingError):
+            cleanup_unused_apps('soa_dir', kill_threshold=0, force=False)
+
+
+def test_cleanup_unused_apps_force():
+    mock_kube_client = MockKubeClient()
+    namespace = 'paasta'
+    mock_kube_client.deployments.create_namespaced_deployment(namespace, fake_deployment)
+    with mock.patch(
+        'paasta_tools.cleanup_kubernetes_jobs.KubeClient',
+        return_value=mock_kube_client,
+        autospec=True,
+    ), mock.patch(
+        'paasta_tools.cleanup_kubernetes_jobs.get_services_for_cluster',
+        return_value={},
+        autospec=True,
+    ), mock.patch(
+        'paasta_tools.cleanup_kubernetes_jobs.notify',
+        autospec=True,
+    ) as mock_notify:
+        mock_notify.__enter__ = mock.Mock(return_value=(mock.Mock(), None))
+        mock_notify.__exit__ = mock.Mock(return_value=None)
+        cleanup_unused_apps('soa_dir', kill_threshold=0, force=True)
+        assert len(mock_kube_client.deployments.list_namespaced_deployment(namespace).items) == 0
+
+
+def test_cleanup_unused_apps_ignore_invalid_apps():
+    mock_kube_client = MockKubeClient()
+    namespace = 'paasta'
+    mock_kube_client.deployments.create_namespaced_deployment(namespace, invalid_app)
+    with mock.patch(
```

Comment:
You should define these mocks as fixtures since they are frequently used. You can even set autouse=True if you don't care about the return value - that would be especially useful for notify so we never forget to mock it.